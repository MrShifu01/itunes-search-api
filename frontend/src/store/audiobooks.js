import { createSlice } from "@reduxjs/toolkit";

const audiobooksSlice = createSlice({
    name: "audiobooks",

    initialState: {
        audiobooksArray: []
    },

    reducers: {
        addAudioBooks: (state, action) => {
            state.audiobooksArray.push(action.payload)
        },

        removeAudioBooks: (state, action) => {
            state.audiobooksArray.splice(action.payload, 1) 
        },

        resetAudioBooks: (state) => {
            state.audiobooksArray = []
        }
    }
})

export const {addAudioBooks, removeAudioBooks, resetAudioBooks} = audiobooksSlice.actions

export default audiobooksSlice.reducer