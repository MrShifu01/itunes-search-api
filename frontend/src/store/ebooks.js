import { createSlice } from "@reduxjs/toolkit";

const ebooksSlice = createSlice({
    name: "ebooks",

    initialState: {
        ebooksArray: []
    },

    reducers: {
        addEbooks: (state, action) => {
            state.ebooksArray.push(action.payload)
        },

        removeEbooks: (state, action) => {
            state.ebooksArray.splice(action.payload, 1) 
        },

        resetEbooks: (state) => {
            state.ebooksArray = []
        }
    }
})

export const {addEbooks, removeEbooks, resetEbooks} = ebooksSlice.actions

export default ebooksSlice.reducer