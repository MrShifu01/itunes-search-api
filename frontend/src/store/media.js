import { createSlice } from "@reduxjs/toolkit";

const mediaSlice = createSlice({
  name: "media", // The name of the slice

  initialState: {
    media: "all", // The initial media value
  },

  reducers: {
    // Reducer function for changing the media value
    changeMedia: (state, action) => {
      state.media = action.payload;
    },
  },
});

// Extract the action creator from the slice
export const { changeMedia } = mediaSlice.actions;

// Export the reducer function from the slice
export default mediaSlice.reducer;
