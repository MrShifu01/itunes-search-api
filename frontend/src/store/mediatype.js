import { createSlice } from "@reduxjs/toolkit";

const mediaTypeSlice = createSlice({
  name: "mediatype", // The name of the slice

  initialState: {
    tvshows: [], // Array to store TV shows
    music: [], // Array to store music
    movies: [], // Array to store movies
    podcasts: [], // Array to store podcasts
    musicvideos: [], // Array to store music videos
    audiobooks: [], // Array to store audiobooks
    software: [], // Array to store software
    ebooks: [], // Array to store ebooks
  },

  reducers: {
    // Reducer functions for adding media to the corresponding arrays
    addTvShows: (state, action) => {
      state.tvshows.push(action.payload);
    },
    addMusic: (state, action) => {
      state.music.push(action.payload);
    },
    addMovies: (state, action) => {
      state.movies.push(action.payload);
    },
    addPodcasts: (state, action) => {
      state.podcasts.push(action.payload);
    },
    addMusicVideos: (state, action) => {
      state.musicvideos.push(action.payload);
    },
    addAudioBooks: (state, action) => {
      state.audiobooks.push(action.payload);
    },
    addSoftware: (state, action) => {
      state.software.push(action.payload);
    },
    addEbooks: (state, action) => {
      state.ebooks.push(action.payload);
    },

    // Reducer functions for removing media from the corresponding arrays
    removeTvShows: (state, action) => {
      state.tvshows.splice(action.payload, 1);
    },
    removeMusic: (state, action) => {
      state.music.splice(action.payload, 1);
    },
    removeMovies: (state, action) => {
      state.movies.splice(action.payload, 1);
    },
    removePodcasts: (state, action) => {
      state.podcasts.splice(action.payload, 1);
    },
    removeMusicVideos: (state, action) => {
      state.musicvideos.splice(action.payload, 1);
    },
    removeAudioBooks: (state, action) => {
      state.audiobooks.splice(action.payload, 1);
    },
    removeSoftware: (state, action) => {
      state.software.splice(action.payload, 1);
    },
    removeEbooks: (state, action) => {
      state.ebooks.splice(action.payload, 1);
    },
  },
});

// Extract the action creators from the slice
export const {
  addTvShows,
  addMusic,
  addMovies,
  addPodcasts,
  addMusicVideos,
  addAudioBooks,
  addSoftware,
  addEbooks,
  removeTvShows,
  removeMusic,
  removeMovies,
  removePodcasts,
  removeMusicVideos,
  removeAudioBooks,
  removeSoftware,
  removeEbooks,
} = mediaTypeSlice.actions;

// Export the reducer function from the slice
export default mediaTypeSlice.reducer;
