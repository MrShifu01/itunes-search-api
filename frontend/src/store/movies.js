import { createSlice } from "@reduxjs/toolkit";

const moviesSlice = createSlice({
    name: "movies",

    initialState: {
        moviesArray: []
    },

    reducers: {
        addMovies: (state, action) => {
            state.moviesArray.push(action.payload)
        },

        removeMovies: (state, action) => {
            state.moviesArray.splice(action.payload, 1) 
        },

        resetMovies: (state) => {
            state.moviesArray = []
        }
    }
})

export const {addMovies, removeMovies, resetMovies} = moviesSlice.actions

export default moviesSlice.reducer