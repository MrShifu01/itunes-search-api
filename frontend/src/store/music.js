import { createSlice } from "@reduxjs/toolkit";

const musicSlice = createSlice({
    name: "music",

    initialState: {
        musicArray: []
    },

    reducers: {
        addMusic: (state, action) => {
            state.musicArray.push(action.payload)
        },

        removeMusic: (state, action) => {
            state.musicArray.splice(action.payload, 1) 
        },

        resetMusic: (state) => {
            state.musicArray = []
        }
    }
})

export const {addMusic, removeMusic, resetMusic} = musicSlice.actions

export default musicSlice.reducer