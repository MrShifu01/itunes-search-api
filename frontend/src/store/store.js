import { configureStore } from "@reduxjs/toolkit";
import resultsReducer from './results';
import mediaTypeReducer from './mediatype';
import mediaReducer from './media';
import pageReducer from './page';

export default configureStore({
  reducer: {
    // Set up the root reducer with multiple reducers
    results: resultsReducer, // Reducer for managing search results
    page: pageReducer, // Reducer for managing the current page
    media: mediaReducer, // Reducer for managing media favorites
    mediatype: mediaTypeReducer, // Reducer for managing media types
  }
});
