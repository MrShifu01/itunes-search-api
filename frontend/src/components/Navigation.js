import '../index.css'
import { useDispatch } from 'react-redux'
import { changePage } from '../store/page';
import { useState } from 'react';

function Navigation() {

// Declaring whether the drop down menu must be open or closed
  const [isDropdownOpen, setIsDropdownOpen] = useState(false)
  const dispatch = useDispatch();

  return (
    <>
        
        {/* Navbar Container */}
        <div className="navbar-container bg-neutral-focus flex justify-between">
        <div className='navbar-content p-3'>
            
            {/* Logo/Home */}
            <div
            className='drop-item cursor-pointer'
            onClick={(e) => {
                e.preventDefault()
                dispatch(changePage('search'))}}
            >favour-A-tism</div>
            
            {/* Drop Down Menu */}
            <div className="dropdown">
                
                <button 
                className="dropbtn"
                onClick={() => setIsDropdownOpen(prev => !prev)}
                >
                <i className='bx bx-chevron-down px-2'></i>Favourites
                </button>
                <div className={`dropdown-content w-40 rounded-md p-3 bg-neutral-focus flex flex-col ${isDropdownOpen ? '' : 'hidden'}`}>
                
                <a
                className='drop-item' 
                href="/"
                onClick={(e) => {
                    e.preventDefault()
                    dispatch(changePage('music'))
                    setIsDropdownOpen(false)
                }}
                >Music</a>
                
                <a 
                className='drop-item'
                href="/"
                onClick={(e) => {
                    e.preventDefault()
                    dispatch(changePage('movie'))
                    setIsDropdownOpen(false)
                }}
                >Movies</a>
                
                <a 
                className='drop-item'
                href="/"
                onClick={(e) => {
                    e.preventDefault()
                    dispatch(changePage('podcast'))
                    setIsDropdownOpen(false)
                }}
                >Podcasts</a>
                
                <a 
                className='drop-item'
                href="/"
                onClick={(e) => {
                    e.preventDefault()
                    dispatch(changePage('musicVideo'))
                    setIsDropdownOpen(false)
                }}
                >Music Videos</a>
                
                <a 
                className='drop-item'
                href="/"
                onClick={(e) => {
                    e.preventDefault()
                    dispatch(changePage('audiobook'))
                    setIsDropdownOpen(false)
                }}
                >Audiobooks</a>
                
                <a 
                className='drop-item'
                href="/"
                onClick={(e) => {
                    e.preventDefault()
                    dispatch(changePage('tvShow'))
                    setIsDropdownOpen(false)
                }}
                >TV Shows</a>
                
                <a 
                className='drop-item'
                href="/"
                onClick={(e) => {
                    e.preventDefault()
                    dispatch(changePage('software'))
                    setIsDropdownOpen(false)
                }}
                >Software</a>
                
                <a 
                className='drop-item'
                href="/"
                onClick={(e) => {
                    e.preventDefault()
                    dispatch(changePage('ebook'))
                    setIsDropdownOpen(false)
                }}
                >eBooks</a>

                </div>
            </div>
        </div>
        </div>
    </>
  );
}

export default Navigation;
