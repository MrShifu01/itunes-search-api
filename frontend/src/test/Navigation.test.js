import renderer from 'react-test-renderer';
import Navigation from '../components/Navigation';
import store from '../store/store';
import { Provider } from 'react-redux';

test('Navigation renders correctly', () => {
    // Create a React test renderer instance
    const tree = renderer
        .create(
            // Wrap the Navigation component with the Provider and provide the store
            <Provider store={store}>
                <Navigation/>
            </Provider>
        )
        // Convert the rendered component to a JSON representation
        .toJSON();
    
    // Compare the rendered component snapshot with the saved snapshot
    expect(tree).toMatchSnapshot();
});