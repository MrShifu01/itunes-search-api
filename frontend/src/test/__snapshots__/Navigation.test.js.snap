// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div
  className="navbar-container bg-neutral-focus flex justify-between"
>
  <div
    className="navbar-content p-3"
  >
    <div
      className="drop-item cursor-pointer"
      onClick={[Function]}
    >
      favour-A-tism
    </div>
    <div
      className="dropdown"
    >
      <button
        className="dropbtn"
        onClick={[Function]}
      >
        <i
          className="bx bx-chevron-down px-2"
        />
        Favourites
      </button>
      <div
        className="dropdown-content w-40 rounded-md p-3 bg-neutral-focus flex flex-col hidden"
      >
        <a
          className="drop-item"
          href="/"
          onClick={[Function]}
        >
          Music
        </a>
        <a
          className="drop-item"
          href="/"
          onClick={[Function]}
        >
          Movies
        </a>
        <a
          className="drop-item"
          href="/"
          onClick={[Function]}
        >
          Podcasts
        </a>
        <a
          className="drop-item"
          href="/"
          onClick={[Function]}
        >
          Music Videos
        </a>
        <a
          className="drop-item"
          href="/"
          onClick={[Function]}
        >
          Audiobooks
        </a>
        <a
          className="drop-item"
          href="/"
          onClick={[Function]}
        >
          TV Shows
        </a>
        <a
          className="drop-item"
          href="/"
          onClick={[Function]}
        >
          Software
        </a>
        <a
          className="drop-item"
          href="/"
          onClick={[Function]}
        >
          eBooks
        </a>
      </div>
    </div>
  </div>
</div>
`;
